src/pip/_internal/commands/__init__.py

```diff
@@ -100,7 +97,8 @@ def create_command(name, **kwargs):
     return command
 
 
-def get_similar_commands(name):
+def get_similar_command(name):
+    # type: (str) -> Optional[Sequence[str]]
```

Comment:
```suggestion
    # type: (str) -> Optional[str]
```

Reply:
Hi @pradyunsg ,

As I clarified https://github.com/pypa/pip/pull/7977#discussion_r403496601 , we cannot use `Optional[str]` and have to use `Optional[Sequence[str]]` . 

Could you please take a look at the explanation, and LMK if something is missing, and I will fix this accordingly

Reply:
Well. Looks like that means difflib's type annotations are incorrect. :)

See https://docs.python.org/2/library/difflib.html#difflib.get_close_matches and https://github.com/python/typeshed/blob/master/stdlib/2and3/difflib.pyi#L42 -- there's a clear mismatch here.

We'd want to fix this bug in mypy/typeshed before updating our own annotations. :)

Reply:
Good catch @pradyunsg , I didn't see that. 

I can file a bug with them mentioning about this. But do you want to merge this PR (I can make a note of why we are using Sequence[str] with the PR filed in a comment)  and if and when it gets fixed there, we can revisit it.

Reply:
Aah, looks like an issue was already filed https://github.com/python/typeshed/issues/2067 and was marked as `false-positive`, so I am not sure if I file a similar bug again, it will get fixed. I can note dow this bug in the PR if need be.

Reply:
Hi, the reason for the `Sequence[str]` is because `str.__getitem__` returns a `str`, so technically `str` is a `Sequence[str]` (ugh how dumb this is).  The solution to this is simply to tell mypy what to see by `cast`ing, i.e.
```python
def get_similar_commands(name):
    # type: (str) -> Optional[str]
    """Command name auto-correct."""
    from difflib import get_close_matches

    name = name.lower()
    close_commands = cast(List[str], get_close_matches(name, commands_dict))
    try:
        return close_commands[0]
    except IndexError:
        return None
```
Again, I want to emphasize that the chance that the mistyped subcommand is not similar to any existing one is rather low, so using a `try` there might makes more sense.

Reply:
Hi @McSinyx 

Referring to @pradyunsg comment here, I have removed the file containing the contentious type annotation, and we will wait for typeshed to fix it in order to address it again

Reply:
With what our BDFL said in python/mypy#1965, I don't think this can be fixed at all (since the issue is with the confusion between `str` and `Sequence[str]`).  Also what pradyunsg said was

> I don't think we should add a misleading type annotation to our codebase.

With proper casting, the type hint is not any longer misleading.

Reply:
I also discussed with @pradyunsg on [irc](http://kafka.dcpython.org/day/pypa-dev/2020-04-04) that we will do (1) Add  another PR have the correct annotating and (2) use `pip._internal.utils.typing.cast`  to cast the value to the correct type, adding a comment that describes why we're doing a cast there, linking to the issue that was filed with typeshed.

Reply:
@deveshks, I'm sorry for not participating in the full conversation.  Keep up the good work!

Reply:
No worries @McSinyx 😄